addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoItemValueConverter.java

```diff
@@ -0,0 +1,149 @@
+                // Without this catch, the whole sitemap, or page can not be displayed!
+                // This also handles IllegalFormatConversionException, which is a subclass of IllegalArgument.
+                try {
+                    itemValue = ((Type) state).format(format);
+                } catch (IllegalArgumentException e) {
+                    final Logger logger = LoggerFactory.getLogger(DeviceService.class);
+                    logger.warn("Exception while formatting value '{}' of item {} with format '{}': {}", state,
+                            channel.getItemName(), format, e);
+                    itemValue = new String("Err");
+                }
+            } else {
+                itemValue = state.toString();
+            }
+        } else {
+            itemValue = state.toString();
+        }
+        return new NeeoItemValue(itemValue);
+    }
+
+    /**
+     * Takes the given <code>formatPattern</code> and replaces it with a analog
+     * String-based pattern to replace all value Occurrences with a dash ("-").
+     *
+     * @param formatPattern the original pattern which will be replaces by a
+     *            String pattern.
+     * @return a formatted String with dashes ("-") as value replacement
+     */
+    private static String formatUndefined(String formatPattern) {
+        String undefinedFormatPattern = formatPattern.replaceAll(IDENTIFY_FORMAT_PATTERN_PATTERN, "%1\\$s");
+        try {
+            return String.format(undefinedFormatPattern, "-");
+        } catch (Exception e) {
+            final Logger logger = LoggerFactory.getLogger(DeviceService.class);
+            logger.warn("Exception while formatting undefined value [sourcePattern={}, targetPattern={}, {}]",
+                    formatPattern, undefinedFormatPattern, e);
+            return "Err";
+        }
+    }
+
+    /**
+     * Convert the {@link Item} and {@link PathInfo} to a {@link Command}. Buttons and switches have special handling,
+     * all others are handled by {@link TypeParser#parseCommand(java.util.List, String)}
+     *
+     * @param item the non-null item
+     * @param eventType the non-null event type
+     * @return the command
+     */
+    public static Command convert(Item item, PathInfo eventType) {
+        Objects.requireNonNull(item, "item cannot be null");
+        Objects.requireNonNull(eventType, "eventType cannot be null");
+
+        switch (eventType.getComponentType()) {
+            case "button":
+                for (Class<? extends Command> cmd : item.getAcceptedCommandTypes()) {
+                    final Command c = NeeoUtil.getEnum(cmd, eventType.getActionValue());
+                    if (c != null) {
+                        return c;
+                    }
+                }
+                break;
+
+            case "switch":
+                if (StringUtils.equalsIgnoreCase("true", eventType.getActionValue())) {
```

Comment:
If one of the arguments to compare is a static string using StringUtils is not needed as it tests if the value is null, but it's static: This can be simplified to: `"true".equalsIgnoreCase(eventType.getActionValue())`

Reply:
This is very true however I prefer to have a single approach to comparing strings rather than two approaches (depending on static strings or not).  To me - having consistency in the code makes it more readable and less error prone